clang/lib/Sema/SemaDecl.cpp

```diff
@@ -8663,9 +8665,12 @@ void Sema::CheckVariableDeclarationType(VarDecl *NewVD) {
   // ISO/IEC TR 18037 S5.1.2
   if (!getLangOpts().OpenCL && NewVD->hasLocalStorage() &&
       T.getAddressSpace() != LangAS::Default) {
-    Diag(NewVD->getLocation(), diag::err_as_qualified_auto_decl) << 0;
-    NewVD->setInvalidDecl();
-    return;
+    if (!getLangOpts().SYCLIsDevice &&
```

Comment:
Can you please explain this check? How is it related to `wg_scope`? Do we need to add a test for it somewhere?